src/uct/base/uct_mem.c

```diff
@@ -99,15 +99,21 @@ ucs_status_t uct_mem_alloc(void *addr, size_t min_length, unsigned flags,
                     return status;
                 }
 
-                /* Check if MD supports allocation */
-                if (!(md_attr.cap.flags & UCT_MD_FLAG_ALLOC)) {
-                    continue;
-                }
-
-                /* Check if MD supports allocation with fixed address
-                 * if it's requested */
-                if ((flags & UCT_MD_MEM_FLAG_FIXED) &&
-                    !(md_attr.cap.flags & UCT_MD_FLAG_FIXED)) {
+                if (!(flags & UCT_MD_MEM_FLAG_DM)) {
+                    /* Check if MD supports allocation */
+                    if (!(md_attr.cap.flags & UCT_MD_FLAG_ALLOC)) {
+                        continue;
+                    }
+
+                    /* Check if MD supports allocation with fixed address
+                     * if it's requested */
+                    if ((flags & UCT_MD_MEM_FLAG_FIXED) &&
```

Comment:
why do we ignore FIXED and ALLOC in case of UCT_MD_MEM_FLAG_DM?
need to clearly document in UCT API which combinations are supported or not

Reply:
FIXED flag is processed in allocation call (DM + FIXED = UNSUPPORTED), ALLOC is just hint which may be ignored, right?